File: src/highest-product-of-three/highest_product_of_three.py
def higest_product_of_three( arr ):
	"""
	**********************************************************************************************************************
	Given an array_of_ints, find the highest_product you can get from three of the integers.
	"""
	## Highest product of three will be either the highest three ##
	## numbers or the highest and the two largest negative numbers ##
	highest = max( arr[0], arr[1] )
	lowest = min( arr[0], arr[1] )
	highest_of_two = highest * lowest
	lowest_of_two = highest * lowest
	max_product = highest_of_two * arr[2]

	for elem in arr:
		max_product = max( max_product, elem * highest_of_two, elem * lowest_of_two )
		highest_of_two = max( highest_of_two, elem * highest, elem * lowest )
		lowest_of_two = min( lowest_of_two, elem * lowest, elem * highest )
		highest = max( highest, elem )
		lowest = min( lowest, elem )

	return max_product

File: src/highest-product-of-three/highest-product-of-three.js
/**********************************************************************************************************************
 * Given an array_of_ints, find the highest_product you can get from three of the integers.
 */
 function highestProductOfThree( arr )
 {
 	/** Highest product of three will be the three highest elements, **/
 	/** or the highest and the two largest negatives **/

 	var highest = Math.max( arr[0], arr[1] );
 	var lowest = Math.min( arr[0], arr[1] );
 	var highestOfTwo = highest * lowest;
 	var lowestOfTwo = highest * lowest;
 	var highestOfThree = highestOfThree * arr[2];

 	arr.forEach( function( elem )
 	{
 		highestOfThree = Math.max( highestOfThree, elem * highestOfTwo, elem * lowestOfTwo );
 		highestOfTwo = Math.max( highestOfTwo, elem * highest, elem * lowest );
 		lowestOfTwo = Math.min( lowestOfTwo, elem * lowest, elem * highest );
 		highest = Math.max( highest, elem );
 		lowest = Math.min( lowest, elem );
 	});
 	return highestOfThree;
 }